ui/raidboss/data/06-ew/trial/rubicante.ts

```diff
@@ -0,0 +1,284 @@
+02 = Middle ring
+03 = Outer ring
+04 = Flamespire brand indicator?
+
+00 flags:
+00020001 = Fiery
+00080004 = Not fiery
+
+01/02/03 flags:
+00020001 = Arrows rotating CW
+00080004 = Clear CW arrows
+00200010 = Arrows rotating CCW
+00400004 = Clear CCW arrows
+
+04 flags:
+00010001 = cardinals safe?
+00200020 = intercards safe?
+00080004 = clear indicator
+ */
+
+// 7CD4 Ghastly Torch during add phase *is* an aoe but is constant and small, so skipped.
+
+// TO-DO:
+//  - Rewrite Ordeal of Purgation to use Map Effects and identify directional source of cone/safe area(s)
+//  - Flamerake - identify initial direction of line cleave?
+
+export interface Data extends RaidbossData {
+  decOffset?: number;
+  dualfireTarget: boolean;
+}
+
+// First headmarker is Shattering Heat (tankbuster on MT)
+const firstHeadmarker = parseInt('0156', 16);
+const getHeadmarkerId = (data: Data, matches: NetMatches['HeadMarker']) => {
+  if (typeof data.decOffset === 'undefined')
+    data.decOffset = parseInt(matches.id, 16) - firstHeadmarker;
+  return (parseInt(matches.id, 16) - data.decOffset).toString(16).toUpperCase().padStart(4, '0');
+};
+
+const triggerSet: TriggerSet<Data> = {
+  zoneId: ZoneId.MountOrdeals,
+  timelineFile: 'rubicante.txt',
+  initData: () => {
+    return {
+      dualfireTarget: false,
+    };
+  },
+  triggers: [
+    {
+      id: 'Rubicante Headmarker Tracker',
+      type: 'HeadMarker',
+      netRegex: {},
+      condition: (data) => data.decOffset === undefined,
+      run: (data, matches) => {
+        getHeadmarkerId(data, matches);
+      },
+    },
+    {
+      id: 'Rubicante Inferno Raidwide',
+      type: 'StartsUsing',
+      netRegex: { id: '7CEA', source: 'Rubicante', capture: false },
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Rubicante Ordeal of Purgation',
+      type: 'StartsUsing',
+      netRegex: { id: ['7CC4', '80E8'], source: 'Rubicante', capture: false },
+      alertText: (_data, _matches, output) => output.avoid!(),
+      outputStrings: {
+        avoid: {
+          en: 'Avoid cone',
+        },
+      },
+    },
+    {
+      id: 'Rubicante Shattering Heat',
+      type: 'StartsUsing',
+      netRegex: { id: '7CEB', source: 'Rubicante' },
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'Rubicante Blazing Rapture',
+      type: 'StartsUsing',
+      netRegex: { id: '7CD2', source: 'Rubicante', capture: false },
+      // This is a ~14 second cast.
+      delaySeconds: 5,
+      response: Responses.bigAoe(),
+    },
+    {
+      id: 'Rubicante Inferno Spread',
+      type: 'StartsUsing',
+      netRegex: { id: '7CDA', source: 'Rubicante' },
+      condition: Conditions.targetIsYou(),
+      response: Responses.spread(),
+    },
+    {
+      id: 'Rubicante Flamerake',
+      type: 'StartsUsing',
+      netRegex: { id: '7CDD', source: 'Rubicante', capture: false },
+      // Spinning compass animation is 6 seconds; damage goes out 2.6 seconds later
+      delaySeconds: 3,
```

Comment:
What about making this trigger happen on the damage for Flamerake 1 (with a suppress) rather than try to guess the timing? (This is my usual strategy, so that when the call happens it is always safe to move, and never too early.)

Reply:
The mech is a little different in normal mode, so I don't think that would give sufficient warning.  Basically, the timing is:

0.0 Starts casting `7CDD` on self (this corresponds to the compass appearing and spinning)
6.0 Uses `7CDD` on self (corresponds to compass disappearing)
6.8 Starts casting `7CDE` on self (short silent cast)
7.3 Uses `7CDE` on self (corresponds to a flash telegraph of a line cleave through mid - orientation depends on compass)
8.6 Uses `7CDF` (aoe damage applied for flash telegraph loc)
9.6 Uses `7CE0` (aoe damage applied for lines on either side moving outward from original loc)
10.6 Uses `7CE1` (aoe damage applied for lines further out moving outward/edge of arena)

If we triggered off the 21 line (instead of the 20 line) for `7CDD`, it would give only 2.6 seconds to dodge the initial damage, which feels pretty tight - and keying off any later abilityId reduces that further.  

Another option though might be to just eliminate delaySeconds and do an 8-second duration, which would leave the warning up until right before the initial damage goes out?

Reply:
Ok, thanks for the explanation.  I think this is fine as is!